app/models/course/discussion/topic.rb

```diff
@@ -9,6 +9,29 @@ class Course::Discussion::Topic < ActiveRecord::Base
 
   accepts_nested_attributes_for :posts
 
+  # Get all topics of type `Course::Assessment::Answer` in the given course.
```

Comment:
these definitions should belong in their own class, not in the global topic class. There is no need for the topic model to be aware of all these acts_as_discussion classes.